pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -0,0 +1,295 @@
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Add creates a new PravegaCluster Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcilePravegaCluster{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("pravegacluster-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource PravegaCluster
+	err = c.Watch(&source.Kind{Type: &pravegav1alpha1.PravegaCluster{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcilePravegaCluster{}
+
+// ReconcilePravegaCluster reconciles a PravegaCluster object
+type ReconcilePravegaCluster struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a PravegaCluster object and makes changes based on the state read
+// and what is in the PravegaCluster.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcilePravegaCluster) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reconcilePeriod := 30 * time.Second
+	reconcileResult := reconcile.Result{RequeueAfter: reconcilePeriod}
+
+	log.Printf("Reconciling PravegaCluster %s/%s\n", request.Namespace, request.Name)
+
+	// Fetch the PravegaCluster instance
+	pravegaCluster := &pravegav1alpha1.PravegaCluster{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, pravegaCluster)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Printf("PravegaCluster %s/%s not found. Ignoring since object must be deleted\n", request.Namespace, request.Name)
+			return reconcileResult, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Printf("failed to get PravegaCluster: %v", err)
+		return reconcileResult, err
```

Comment:
Since err is not nil, return reconcile.Result{}, err is enough.

Reply:
Right, I'll update it.